File: Java/deltav/src/main/java/com/ithit/webdav/samples/deltavservlet/websocket/NotificationEncoder.java
package com.ithit.webdav.samples.deltavservlet.websocket;

import javax.websocket.EncodeException;
import javax.websocket.Encoder;
import javax.websocket.EndpointConfig;

/**
 * Encodes notification object to the JSON
 */
public class NotificationEncoder implements Encoder.Text<WebSocketServer.Notification> {

    @Override
    public void init(EndpointConfig config) {
    }

    @Override
    public void destroy() {
    }

    @Override
    public String encode(WebSocketServer.Notification notification) throws EncodeException {
        String target = "";
        if (notification instanceof WebSocketServer.MovedNotification) {
            target = "\"targetPath\" : \"" + ((WebSocketServer.MovedNotification) notification).getTargetPath() + "\" ,";
        }
        return "{" + target +
                "\"itemPath\" : \"" + notification.getItemPath() + "\" ," +
                "\"eventType\" : \"" + notification.getOperation() + "\"" +
                "}";
    }
}

File: Kotlin/filesystemstorage/src/main/kotlin/com/ithit/webdav/samples/fsstorageservlet/websocket/NotificationEncoder.kt
package com.ithit.webdav.samples.fsstorageservlet.websocket

import javax.websocket.EncodeException
import javax.websocket.Encoder
import javax.websocket.EndpointConfig

/**
 * Encodes notification object to the JSON
 */
class NotificationEncoder : Encoder.Text<WebSocketServer.Notification> {

    override fun init(config: EndpointConfig) {}

    override fun destroy() {}

    @Throws(EncodeException::class)
    override fun encode(notification: WebSocketServer.Notification): String {
        var target = ""
        if (notification is WebSocketServer.MovedNotification) {
            target = "\"targetPath\" : \"" + notification.targetPath + "\" ,"
        }
        return "{" + target +
                "\"itemPath\" : \"" + notification.itemPath + "\" ," +
                "\"eventType\" : \"" + notification.operation + "\"" +
                "}"
    }
}
